mui/dockwidgets/state_list_widget.py

```diff
@@ -101,7 +132,9 @@ def _create_item(self, state: StateDescriptor) -> QTreeWidgetItem:
         """Creates a new item that represents a given state"""
 
         parent = self._get_state_list(state)
-        return QTreeWidgetItem(parent, [f"State {state.state_id}"])
+        item = QTreeWidgetItem(parent, [f"State {state.state_id}"])
+        item.setData(0, StateListWidget.STATE_ID_ROLE, state.state_id)
```

Comment:
I'm not too familiar with QT's `setData` or best practices, but it looks like `0` is used as an index. Maybe an integer enum to use as the index would be clearer?
```python
@enum.unique()
class StateDataIndex(enum.IntEnum):
    ID = 0
```
or something (feel free to change the above)

Reply:
The `0` here is the column index for that given item row and `StateListWidget.STATE_ID_ROLE` is a constant I defined to be the role/index for the state id data for that column.

Since the current widget only has one column, I have yet to make the column index a constant, but I'm happy to change that if it makes the code clearer. Also, `STATE_ID_ROLE` is currently a class constant. Do you think a separate enum class would work better here?

Thanks for all the helpful feedback!

Reply:
Thank you for the explanation! 

Okay, I think naming the column index as a constant variable for now would make it clearer when we write and read the value. (An enum is probably premature optimization, but if we add another piece of data to another index, we should probably combine them into an enum).

I think `STATE_ID_ROLE` is fine as a class constant and nice that you named it that way. I don't think it needs to be an enum

Reply:
The column index is now a constant